File: SWEA/D3/1209. ［S／W 문제해결 기본］ 2일차 － Sum/［S／W 문제해결 기본］ 2일차 － Sum.py
for _ in range (10):
    tc = int(input())

    arr = [list(map(int, input().split())) for _ in range(100)]

    row_sum = [0] * 100 # 행들의 합 리스트
    col_sum = [0] * 100 # 열들의 합 리스트
    dia_sum = 0 # 대각선의 합
    dia2_sum = 0 # 대각선2의 합

    # 가로
    for i in range(100):
        for j in range(100):
            row_sum[i] += arr[i][j]

    # 세로
    for j in range(100):
        for i in range(100):
            col_sum[j] += arr[i][j]

    # 대각선
    for x in range(100):
        dia_sum += arr[x][x]

    # 대각선2
    for y in range(99, -1, -1):
        dia2_sum += arr[y][99-y]

    answer = []
    answer.extend(row_sum)
    answer.extend(col_sum)
    answer.append(dia_sum)
    answer.append(dia2_sum)

    # 최대값 구하기
    result = max(answer)
    print(f'#{tc} {result}')

File: SWEA/D3/1209. ［S／W 문제해결 기본］ 2일차 － Sum/［S／W 문제해결 기본］ 2일차 － Sum.java
import java.util.Scanner;

public class Solution {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		for (int tc = 1; tc <= 10; tc++) {
			sc.nextInt();

			// 배열 입력
			int[][] matrix = new int[100][100];
			for (int i = 0; i < 100; i++) {
				for (int j = 0; j < 100; j++) {
					matrix[i][j] = sc.nextInt();
				}
			}

			int answer = 0;

			// 가로줄, 세로줄 합
			for (int i = 0; i < 100; i++) {
				int rowSum = 0;
				int colSum = 0;
				for (int j = 0; j < 100; j++) {
					rowSum += matrix[i][j];
					colSum += matrix[j][i];
				}
				int tempMax = Math.max(rowSum, colSum);
				answer = Math.max(answer, tempMax);
			}

			// 대각선 2줄
			int rightSum = 0;
			int leftSum = 0;
			for (int k = 0; k < 100; k++) {
				rightSum += matrix[k][k];
				leftSum += matrix[k][99-k];
			}
			int tempMax2 = Math.max(rightSum, leftSum);
			answer = Math.max(answer, tempMax2);
			
			System.out.printf("#%d %d\n", tc, answer);
			
			
		}//tc
		sc.close();
	}// main
}// class Sum
